src/peft/tuners/adaption_prompt/model.py

```diff
@@ -65,7 +65,7 @@ def add_adapter(self, adapter_name: str, config: AdaptionPromptConfig) -> None:
 
         parents = []
         for name, _ in self.model.named_modules():
-            if name.endswith(config.target_modules):
+            if name.endswith(config.target_modules) and not name.endswith(f"_{config.target_modules}"): #accounts for attn vs c_attn
```

Comment:
Would this not be sufficient?

```suggestion
            if name.endswith(f".{config.target_modules}"):
```

Reply:
I'm not quite happy with this myself, maybe you have a suggestion to deal with the layer naming conventions. If we do as suggested in the GPT2 model, we would end up attempting to apply the AdaptionPrompt to the 1DConv Layer named c_attn, which is a sublayer of the self attention layer "attn" in GPT2.

Reply:
Would my solution not solve the issue? If the module names are `foo.bar.attn` and `bar.baz.c_attn`, then if we check for `".attn"`, we would not accidentally match `c_attn`.